lib/util/host-rules.ts

```diff
@@ -153,6 +153,14 @@ export function find(search: HostRuleSearch) {
   return res;
 }
 
+export function getPlatformByHostOrUrl(url){
+  let {host} = URL.parse(url);
+  for(let rule of hostRules){
+    if(rule.baseUrl == url || rule.hostName == host || rule.domainName == host){
+      return rule
```

Comment:
This is returning a rule, not a platform. What exactly is the requirement?

Reply:
Initially I was thinking to just return the platform on the basis of the sourceurl, but I required a similar function to also verify that there is a rule for a particular url.
So I decided it will be more convenient to just return the rule.
I can ename the function to **getRulesetByHostOrUrl** if you wish?